### PR TITLE
When using Changelog it opens a 'subl' editor and waits for you prompt

### PR DESCRIPTION
**Pull request by hidroo**
Hello,

I wrote some code so when you use `--changelog` it opens the file with `subl` and waits for your confirmation. In this way you do not need to use `git commit --amend` in order to add the CHANGELOG.md.

### CODE REVIEW

**Comment by hidroo** [comment]
The tests are not going to pass, because we wait for a "Y" input. And also because there may not be 'subl' installed on the machine.

If you tell me how to fix that issue I will. I looked at the tests, but could not figure out how to give the test 'Y' input.

File: bump.gemspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 Gem::Specification.new "bump" do |s|
-  s.version = "0.8.0"
+  s.version = "0.8.1"
```
Comment (grosser) [inline bump.gemspec]: don't bump version in PR

File: lib/bump.rb [hunk 2]
```diff
@@ -145,6 +145,8 @@ def bump(file, current, next_version, options)
           error = bump_changelog(log, next_version)
           return [error, 1] if error
 
+          open_changelog(options, log)
```
Comment (grosser) [inline lib/bump.rb]:
```suggestion
        if editor = options[:editor]
          open_changelog(editor, log)
        end
```

File: lib/bump.rb [hunk 3]
```diff
@@ -155,6 +157,21 @@ def bump(file, current, next_version, options)
         ["Bump version #{current} to #{next_version}", 0]
       end
 
+      def open_changelog options, log
+        if options[:editor]
+          return ["Could not find '#{options[:editor]}'", 1] unless command_exists?(options[:editor])
```
Comment (grosser) [inline lib/bump.rb]: I'd say just start it and let it crash 🤷‍♂

File: lib/bump.rb [hunk 4]
```diff
@@ -155,6 +157,21 @@ def bump(file, current, next_version, options)
...
+          system(options[:editor], log)
+          puts "Ready with your Changelog? Y/n"
```
Comment (grosser) [inline lib/bump.rb]: if the editor exited successfully then we should be good to go, just like `git commit` works 🤷‍♂

Comment (hidroo) [reply lib/bump.rb]: The editor can be sublime, vscode, gedit or something else that won't open in the terminal. That is why I want to wait for it.

Comment (grosser) [reply lib/bump.rb]: https://stackoverflow.com/questions/22631863/setting-up-sublimetext-3-as-git-commit-text-editor

if git does not need to support fanciness like that, then we don't either :trollface:

Comment (grosser) [reply lib/bump.rb]: ... also again, best to use `ENV['EDITOR']` 🤷‍♂

Comment (hidroo) [reply lib/bump.rb]: Okay I will look at how to use sublime from your example tomorrow.

What will happen if you have the flag `--editor`, but do not have `ENV['EDITOR']`

Comment (hidroo) [reply lib/bump.rb]: I saw the sublime example. I do not know how to open the editor wait for it to complete...
Do you have a tip?

Comment (grosser) [reply lib/bump.rb]: `EDITOR="subl -n -w"` should do it ... if none is set, fall back to `vi`

Comment (grosser) [reply lib/bump.rb]: flag could be `--edit-changelog` which then also sets `changelog` to true

Comment (hidroo) [reply lib/bump.rb]: `system("subl -n -w", "file")` does not work for me. Could you check it?

``nano file`` does not work eider..

I am thinking of adding an if to check for vi and nano in order to use ```` and `system()` for subl.

File: bin/bump [hunk 5]
```diff
@@ -28,6 +28,7 @@ OptionParser.new do |opts|
   opts.on("--tag-prefix TAG_PREFIX", "Prefix the tag with this string, ex. 'v'") { |tag_prefix| options[:tag_prefix] = tag_prefix }
   opts.on("--replace-in FILE", String, "Replace old version with the new version additionally in this file") { |f| (options[:replace_in] ||= []) << f }
   opts.on("--changelog", "Update CHANGELOG.md") { options[:changelog] = true }
+  opts.on("--edit-changelog", "Use an editor from ENV['EDITOR'] to open changelog. ex. 'subl -n -w' or 'nano'.") { options[:editor] = true; options[:changelog] = true }
```
Comment (grosser) [inline bin/bump]:
```suggestion
  opts.on("--edit-changelog", "Use an $EDITOR to open changelog, e.g. 'subl -n -w' or 'nano'.") { options[:changelog] = :editor }
```

File: lib/bump.rb [hunk 6]
```diff
@@ -155,6 +157,21 @@ def bump(file, current, next_version, options)
...
       end
 
+      def open_changelog(log)
+        editor = ENV['EDITOR'] || "vi"
+
+        if editor.include?("subl")
+          `#{editor} #{log}`
+        else
+          system(editor, log)
+        end
```
[the remark this answers is not in the document]
Comment (grosser) [reply lib/bump.rb]: try `system "#{editor} #{log}"`

**Comment by grosser** [comment]
looks really simple/clean now :)

**Comment by hidroo** [comment]
Are we gonna write tests about this? I do not know how to mock the editor. Do you have any ideas?

**Comment by hidroo** [comment]
Hello, what are the steps now? Are you
Going to merge it?

**Comment by grosser** [comment]
I'll give it a try and add some tests :)

**Comment by grosser** [comment]
release coming soon ™️

**Pull request merged**